File: src/Goods.Domain/Regions/Regions.cs
namespace Goods.Domain.Products;

public class Regions
{
    public Guid Id { get; set; }
    public String Name { get; set; }
    public FederalRegions FederalRegion { get; set; }
    public Regions
    (
        Guid id,
        String name,
        FederalRegions federalRegion
    )
    {
        Id = id;
        Name = name;
        FederalRegion = federalRegion;
    }
}

File: src/Goods.BackOffice/client-app/src/domain/regions/regions.ts
import { Page } from '../../tools/types/page';
import { FederalRegions } from '../federalRegions/federalRegions';

export class Regions {
	constructor(
		public readonly id: string,
		public readonly name: string,
		public readonly federalRegion: FederalRegions
	) { }
}

export function mapToRegionsPage(data: any): Page<Regions> {
	return Page.convert(data, mapToRegion);
}

export function mapToRegions(data: any[]): Regions[] {
	return data.map(mapToRegion);
}

export function mapToRegion(data: any): Regions {
	return new Regions(data.id, data.name, data.federalRegion);
}
